sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -598,7 +598,10 @@ static int sof_audio_probe(struct platform_device *pdev)
 	if (!ctx)
 		return -ENOMEM;
 
-	dmi_check_system(sof_rt5682_quirk_table);
+	if (pdev->id_entry && pdev->id_entry->driver_data)
+		sof_rt5682_quirk = (unsigned long)pdev->id_entry->driver_data;
+	else
```

Comment:
I would have removed the else case and do the dmi check unconditionally.

Reply:
Sure @plbossart . I have removed the else condition too. This will allow dmi matching method to override even if it is in platform driver table.
Updated patchset.